server/chat/backend/agent/agent.py

```
@@ -783,6 +812,66 @@ def on_llm_end(self, response, run_id=None, **kwargs):
+                                                    tc_name = tc.get('name') if isinstance(tc, dict) else getattr(tc, 'name', None)
+                                                    tc_args = tc.get('args') if isinstance(tc, dict) else getattr(tc, 'args', {})
+                                                    if not tc_id or not tc_name or tc_args is None:
+                                                        continue
+                                                    # Register into current_tool_calls so the
+                                                    # cloud_tools wrapper can match this call
+                                                    # on completion and fire capture_tool_end.
+                                                    # Without this, sub-agent execution_steps
+                                                    # rows stay status='running' forever.
+                                                    try:
+                                                        tool_capture.capture_tool_start(tc_name, tc_args, tool_call_id=tc_id)
+                                                    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)